include/uk/arch/types.h

```diff
@@ -257,9 +257,15 @@ typedef struct {
 	__u32 counter;
 } __atomic;
 
+#ifdef __cplusplus
+typedef bool __bool;
```

Comment:
Am I missing something or this will actually not work on C++ without a standard library defining `bool`? To me it looks like we are introducing a dependency to libc on kernel code that uses `__bool`, because of compatibility with C++.

Reply:
Isn't `bool` a basic cpp type? Why would it need a standard library?

Reply:
Yup, my bad. `bool` is indeed, `_Bool` is not.